### ISSUE TITLE
Exclusive mode for rows&columns in Switch matrix modules

### ISSUE THREAD

**Issue by vcvrackideas** [bogaudio/BogaudioModules#112]
Hi! I think this would be very useful options for those who don't want to use your Switch matrix modules as a mixer but as a multichannel switches. It would be also helpful to route 1v\oct signals for example for preventing to mix those ones, only one in row or column (depends on possible options of course, two separate options would be cool) can appears.

**Comment by mdemanett** [comment]
I like this idea, and went ahead and implemented it....

Note it does not work well with midi mapping: the midi mapper (the Rack built in one) constantly resets the mapped control's value to the midi controller's value, and this overwrites the changes the "exclusivity" tries to make.  Example: set "exclusive by rows" on, then midi map a button in a row, and toggle it on with the controller, then click a different button in the row: both buttons are on.  I don't see a solution to this.

**Comment by vcvrackideas** [comment]
Thanks a lot! Actually I got an idea from multi-toggle set of buttons in Touchosc (it has exclusive option per one instance) and that worked with Midi cat from stoermelder well due to proper midi feedbacking. Gonna test it with the new feature soon

**Comment by vcvrackideas** [comment]
Confirmed! - Fully working with TouchOsc toggle buttons even without internal touchoscs exclusive features, both for row and column via Midi-Cat stoermelder's and I got a super proper midi feedback, no lags glitches or so. It seems to me it automatically deselect previous button according to current state of switch and then switch to the another one. This is so awesome and huge improvement for those who want to control patching connections even without a mouse with prepared template based on your coolest matrix. Thanks again Matt!!!
